pkg/apis/networking/validation/validation_test.go

```diff
@@ -33,1022 +33,493 @@ import (
-							{
-								PodSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-					},
-				},
+			{
+				Protocol: &protocolUDP,
+				Port:     &intstr.IntOrString{Type: intstr.String, StrVal: "dns"},
 			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Ingress: []networking.NetworkPolicyIngressRule{
-					{
-						From: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-					},
-				},
+			{
+				Protocol: &protocolSCTP,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 7777},
 			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Ingress: []networking.NetworkPolicyIngressRule{
-					{
-						From: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-								PodSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"e": "f"},
-								},
-							},
-						},
-					},
-				},
+		}
+	}
+
+	setIngressPortsHigher := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].Ports = []networking.NetworkPolicyPort{
+			{
+				Protocol: &protocolTCP,
+				Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 32768},
+				EndPort:  &endPort,
 			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{
-					MatchLabels: map[string]string{"a": "b"},
-				},
-				Egress: []networking.NetworkPolicyEgressRule{
-					{
-						To: []networking.NetworkPolicyPeer{
-							{
-								NamespaceSelector: &metav1.LabelSelector{
-									MatchLabels: map[string]string{"c": "d"},
-								},
-							},
-						},
-					},
-				},
-				Ingress: []networking.NetworkPolicyIngressRule{
-					{
-						From: []networking.NetworkPolicyPeer{
-							{
-								IPBlock: &networking.IPBlock{
-									CIDR:   "192.168.0.0/16",
-									Except: []string{"192.168.3.0/24", "192.168.4.0/24"},
-								},
-							},
-						},
-					},
-				},
+		}
+	}
+
+	setIngressFromPodSelector := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].From[0].PodSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{"c": "d"},
+		}
+	}
+
+	setAlternativeIngressFromPodSelector := func(networkPolicy *networking.NetworkPolicy) {
+		networkPolicy.Spec.Ingress[0].From[0].PodSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{"e": "f"},
+		}
+	}
```

Comment:
In cases like this, does it make more sense to do:

```
setIngressFromPodSelector := func(k, v, string) func(*networking.NetworkPolicy) {
    return func(networkPolicy *networking.NetworkPolicy) {
￼		networkPolicy.Spec.Ingress[0].From[0].PodSelector = &metav1.LabelSelector{
￼			MatchLabels: map[string]string{k: v},
￼		}
￼	}
}
```

Then call it as `setIngressFromPodSelector("c", "d")` or `setIngressFromPodSelector("e", "f")`.  In this way you can sort of build up a mini functional DSL.

Even for cases where you don't have multiple identical-but-for-args implementations, this can make the test cases clearer - the goal is to make each test-case as obvious and terse as possible, and not terser.